server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -623,6 +627,23 @@ private void checkDepthLimit(String objectPath, long maxDepth) {
         }
     }
 
+    private void checkFieldNameSoftLimit(Collection<ObjectMapper> objectMappers,
+                                         Collection<FieldMapper> fieldMappers,
+                                         Collection<FieldAliasMapper> fieldAliasMappers) {
+        final long maxFieldNameLength = indexSettings.getValue(INDEX_MAPPING_FIELD_NAME_LENGTH_LIMIT_SETTING);
+
+        Stream.of(objectMappers.stream(), fieldMappers.stream(), fieldAliasMappers.stream())
+            .reduce(Stream::concat)
+            .orElseGet(Stream::empty)
+            .forEach(mapper -> {
+                String name = mapper.simpleName();
+                if (name.length() > maxFieldNameLength) {
+                    throw new IllegalArgumentException("Field name [" + name + "] is too long; limit is length "
```

Comment:
Maybe rephrase: "the limit is set to ... characters but was..."

Reply:
Can you also add the index name to the exception message? You can get it through `index().getName()`.